### PR TITLE
Support mixed case consul service tags on consul storage engine

### LINKED ISSUE #6427: Storage consul: upper case letters in service_tags are lowered
**Describe the bug**
Value set in consul tags should been stricly the same that defined in `service_tags` consul storage config.

**To Reproduce**

1. Setup a consul backend storage with upper case : 
```hcl
storage "consul" {
  address = "<...>:8500"
  path    = "vault"
  service = "vault"
  service_tags = "traefik.enable=true,traefik.frontend.rule=Host:test-vault.mydomain.com"
  token   = "<...>"
}
```
2. Restart / Unseal your vault
3. Check on Consul UI your `vault` service, the `Host` define in vault configuration became `host`

**Expected behavior**
All upper case in `service_tags` should not been converted in low case.

**Environment:**
* Vault Server Version (retrieve with `vault status`):
```
Key                    Value
---                    -----
Seal Type              shamir
Initialized            true
Sealed                 false
Total Shares           5
Threshold              3
Version                1.0.3
Cluster Name           vault-cluster-<...>
Cluster ID             <...>
HA Enabled             true
HA Cluster             https://<...>:8201
HA Mode                standby
Active Node Address    http://<...>:8200
```

* Vault CLI Version (retrieve with `vault version`):
```
Vault v1.0.3 ('85909e3373aa743c34a6a0ab59131f61fd9e8e43')
```

* Server Operating System/Architecture:
```
Centos 7
Docker 18.09.3 
Official vault image vault:1.0.3
```

Vault server configuration file(s):

```hcl
storage "consul" {
  address = "<...>:8500"
  path    = "vault"
  service = "vault"
  service_tags = "traefik.enable=true,traefik.frontend.rule=Host:test-vault.mydomain.com"
  token   = "<...>"
}
listener "tcp" {
  address           = "0.0.0.0:8200"
  cluster_address   = "0.0.0.0:8201"
  tls_disable       = true
}
default_lease_ttl = "1h"
ui = true
api_addr = "http://<...>:8200"
cluster_addr = "https://<...>:8201"
```

**Additional context**
Theses tags are used by a reverse proxy / load balancer [TRAEFIK](https://traefik.io/) which is case sensitive.

### PR DESCRIPTION
**Pull request by ncabatoff**
When support for service tags was added, the only way we had to parse
and dedup a list of strings also forced them to be lowercase.  Now there's
another helper func that doesn't smash the case so use that instead.

Fixes #6427.

### CODE REVIEW

**Comment by calvn** [comment]
Are there any backward-compat implications in here? `strutil. ParseDedupAndSortStrings` would not dedup `foo` and `Foo` and will show as two separate service tags after this change instead of just `foo`. 

This might be ok, since the "old" tag is still present (in addition to the new one that would no longer be dedupped), just something worth noting since it's a behavioral change.

**Comment by jefferai** [comment]
> When support for service tags was added, the only way we had to parse
and dedup a list of strings also forced them to be lowercase.

I don't think that's the root of the problem, I think it's that when it was coded (by a Consul/TFE dev who contributed this to us) they assumed service tags would always be lowercase.

**Comment by catsby** [comment]
> Are there any backward-compat implications in here? strutil. ParseDedupAndSortStrings would not dedup foo and Foo and will show as two separate service tags after this change instead of just foo.

Is that a mis-configuration that we are incorrectly de-duping into a single tag today?

**Comment by tyrannosaurus-becks** [comment]
In the `ldaputil`, there's a field called `"case_sensitive_names"`. I wonder if we could do something here to make this backwards compatible but support both use cases. Perhaps it could be called something like "case_sensitive_service_tags" and default to `true`.

**Comment by banks** [comment]
I've asked a question on the issue to see if this is still something pressing enough to address (at least the original case folks were running into with Traeffik may no longer be an issue according to their docs at least).

We can decide if we want to merge this anyway - if it's not actively impacting anyone any more we could leave that, but I do think it's better to not impose that restriction on case here in general.

Personally I feel like it would be sufficient to just make this change and note it in the upgrade docs. My _feeling_ (which is 100% pure wild speculation) is that it's probably unlikely enough to cause a major issue for users relying on their supplying upper case tags and having them converted to lower 🤷 . It certainly feels disproportionate to go through a lot more bother of making it backwards compatible not to mention the long-term added complexity of supporting optional modes.

My two cents anyway!

**Comment by banks** [comment]
Seems from the issue this is still a blocker for Correct Traefik routing at least. I think we just need to note the behaviour change in upgrade docs and merge.

Anyone disagree?

File: changelog/6483.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+physical: Consul service registration tags are now case-sensitive.
```
Comment (ncabatoff) [inline changelog/6483.txt:2]:
```suggestion
storage/consul: Consul service registration tags are now case-sensitive.
```

**Review by peteski22 — APPROVED** [review]

**Pull request merged**